contracts/modules/validators/K1Validator.sol

```diff
@@ -182,4 +257,34 @@ contract K1Validator is ERC7739Validator, IERC1271Legacy {
         }
         return size > 0;
     }
+
+    // @notice Fills the safeSenders list from the given data
+    function _fillSafeSenders(bytes calldata data) private {
+        for (uint256 i; i < data.length / 20; i++) {
+            safeSenders.add(msg.sender, address(bytes20(data[20*i:20*(i+1)])));
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                     METADATA
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Returns the name of the module
+    /// @return The name of the module
+    function name() external pure returns (string memory) {
+        return "K1Validator";
+    }
+
+    /// @notice Returns the version of the module
+    /// @return The version of the module
+    function version() external pure returns (string memory) {
+        return "1.0.0-beta";
+    }
+
+    /// @notice Checks if the module is of the specified type
+    /// @param typeId The type ID to check
+    /// @return True if the module is of the specified type, false otherwise
+    function isModuleType(uint256 typeId) external pure returns (bool) {
+        return typeId == MODULE_TYPE_VALIDATOR || typeId == ERC7579_MODULE_TYPE_STATELESS_VALIDATOR;
```

Comment:
ERC7579_MODULE_TYPE_STATELESS_VALIDATOR is this a new accepted type?
should it be part of supportsModule?

Reply:
Yes, it is planned to release it as 7579 extension

But you're right, I think I'll remove it from Nexus as of now